src/engine/blocks.js

```diff
@@ -1,496 +1,501 @@
 
-/**
- * Encode all of `this._blocks` as an XML string usable
- * by a Blockly/scratch-blocks workspace.
- * @return {string} String of XML representing this object's blocks.
- */
-Blocks.prototype.toXML = function () {
-    var xmlString = '<xml xmlns="http://www.w3.org/1999/xhtml">';
-    for (var i = 0; i < this._scripts.length; i++) {
-        xmlString += this.blockToXML(this._scripts[i]);
+    /**
+     * Encode all of `this._blocks` as an XML string usable
+     * by a Blockly/scratch-blocks workspace.
+     * @return {string} String of XML representing this object's blocks.
+     */
+    toXML () {
+        let xmlString = '<xml xmlns="http://www.w3.org/1999/xhtml">';
+        for (let i = 0; i < this._scripts.length; i++) {
+            xmlString += this.blockToXML(this._scripts[i]);
+        }
+        return `${xmlString}</xml>`;
     }
-    return xmlString + '</xml>';
-};
 
-/**
- * Recursively encode an individual block and its children
- * into a Blockly/scratch-blocks XML string.
- * @param {!string} blockId ID of block to encode.
- * @return {string} String of XML representing this block and any children.
- */
-Blocks.prototype.blockToXML = function (blockId) {
-    var block = this._blocks[blockId];
-    // Encode properties of this block.
-    var tagName = (block.shadow) ? 'shadow' : 'block';
-    var xy = (block.topLevel) ?
-        ' x="' + block.x + '" y="' + block.y + '"' :
-        '';
-    var xmlString = '';
-    xmlString += '<' + tagName +
-        ' id="' + block.id + '"' +
-        ' type="' + block.opcode + '"' +
-        xy +
-        '>';
-    // Add any mutation. Must come before inputs.
-    if (block.mutation) {
-        xmlString += this.mutationToXML(block.mutation);
-    }
-    // Add any inputs on this block.
-    for (var input in block.inputs) {
-        if (!block.inputs.hasOwnProperty(input)) continue;
-        var blockInput = block.inputs[input];
-        // Only encode a value tag if the value input is occupied.
-        if (blockInput.block || blockInput.shadow) {
-            xmlString += '<value name="' + blockInput.name + '">';
-            if (blockInput.block) {
-                xmlString += this.blockToXML(blockInput.block);
+    /**
+     * Recursively encode an individual block and its children
+     * into a Blockly/scratch-blocks XML string.
+     * @param {!string} blockId ID of block to encode.
+     * @return {string} String of XML representing this block and any children.
+     */
+    blockToXML (blockId) {
+        const block = this._blocks[blockId];
+        // Encode properties of this block.
+        const tagName = (block.shadow) ? 'shadow' : 'block';
+        const xy = (block.topLevel) ?
+            ` x="${block.x}" y="${block.y}"` :
+            '';
+        let xmlString = '';
+        xmlString += `<${tagName
+            } id="${block.id}"` +
+            ` type="${block.opcode}"${
+            xy
+            }>`;
```

Comment:
The formatting got weird here RE: newlines. Looks like there's a `+` that could be eliminated, too.
I noticed a similar newline issue further down in this file, too.